### PR TITLE
Adding OSD in the distribution build pipeline with build script using volta to install nodejs dependencies

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Make tweaks to dashboards plugin build script to support windows

### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/2306

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: jenkins/check-for-build.jenkinsfile [hunk 1]
```diff
@@ -18,12 +18,12 @@ pipeline {
             H 1 * * * %INPUT_MANIFEST=1.3.7/opensearch-dashboards-1.3.7.yml;TARGET_JOB_NAME=distribution-build-opensearch-dashboards;BUILD_PLATFORM=linux
             H 1 * * * %INPUT_MANIFEST=1.4.0/opensearch-1.4.0.yml;TARGET_JOB_NAME=distribution-build-opensearch;BUILD_PLATFORM=linux macos
             H 1 * * * %INPUT_MANIFEST=1.4.0/opensearch-dashboards-1.4.0.yml;TARGET_JOB_NAME=distribution-build-opensearch-dashboards;BUILD_PLATFORM=linux
-            H 1 * * * %INPUT_MANIFEST=2.3.1/opensearch-2.3.1.yml;TARGET_JOB_NAME=distribution-build-opensearch;BUILD_PLATFORM=linux macos
-            H 1 * * * %INPUT_MANIFEST=2.3.1/opensearch-dashboards-2.3.1.yml;TARGET_JOB_NAME=distribution-build-opensearch-dashboards;BUILD_PLATFORM=linux
+            H 1 * * * %INPUT_MANIFEST=2.3.1/opensearch-2.3.1.yml;TARGET_JOB_NAME=distribution-build-opensearch;BUILD_PLATFORM=linux macos windows
+            H 1 * * * %INPUT_MANIFEST=2.3.1/opensearch-dashboards-2.3.1.yml;TARGET_JOB_NAME=distribution-build-opensearch-dashboards;BUILD_PLATFORM=linux windows
```
Comment (prudhvigodithi) [inline jenkins/check-for-build.jenkinsfile:22]: @peterzhuamazon are we going to 2.3.1 ?
@bbarani

**Comment by prudhvigodithi** [comment]
Hey @peterzhuamazon can the `volta install` be part of the agent, meaning can we pre-bake them?

File: jenkins/check-for-build.jenkinsfile (hunk 1 above)
Comment (peterzhuamazon) [reply jenkins/check-for-build.jenkinsfile:22]: We are planning 2.3.1 if needed thus still running the builds.

**Comment by peterzhuamazon** [comment]
> Hey @peterzhuamazon can the `volta install` be part of the agent, meaning can we pre-bake those packages to the agent?

We already did that but just like container there are issues:

1. Volta, just like NVM is using symlink during installation, which does not preserve after baking into an AMI. The better thing on Volta than NVM is that Volta is using a fixed path and automatically replace binaries on the fly during the build. 

2. A little confusion you may have is Volta does not have a `use` command, thus the `volta install` command here is simply switching the version that we already installed. Remember this image requires to have multiple versions of nodejs so 1.x/2.x/3.x can all be built.

3. Install on the fly also allow us to reduce the number of AMIs to be generated just because a simple nodejs version change, which requires a redeploy of Jenkins to consume the new AMI.

Thanks.

**Comment by tianleh** [comment]
Have we tested the change in a Jenkins job?

**Comment by peterzhuamazon** [comment]
> Have we tested the change in a Jenkins job?

Yes we have tested the PR with 2.4.0 builds:
```

2022-11-03 01:50:01 INFO     Installed plugins: ['alertingDashboards', 'anomalyDetectionDashboards', 'customImportMapDashboards', 'ganttChartDashboards', 'indexManagementDashboards', 'notificationsDashboards', 'observabilityDashboards', 'queryWorkbenchDashboards', 'reportsDashboards', 'searchRelevanceDashboards', 'securityAnalyticsDashboards', 'securityDashboards']

2022-11-03 01:54:18 INFO     Published C:\Users\Administrator\jenkins\workspace\zhujiaxi\bundle-build-zhujiaxi\zip\dist\opensearch-dashboards\opensearch-dashboards-2.4.0-windows-x64.zip.
```

**Review by tianleh — APPROVED** [review]

**Pull request merged**